src/webgl/uniforms.js

```diff
@@ -134,6 +134,27 @@ const UNIFORM_SETTERS = {
   /* eslint-enable max-len */
 };
 
+function toFloatArray(value) {
+  if (value instanceof Float32Array || Array.isArray(value)) {
+    return value;
+  }
+  return Float32Array.from(value);
```

Comment:
1. A possible optimization is to keep a couple of global arrays allocated so we just need to copy into them, rather than create new objects in this case. 

2. I assume this case would be quite rare (it covers initializing a uniform with a different type of typed array?). A good question is if this should even be done automatically (as it can lead to information loss). We chould just assert / throw an exception here and require the application to explicitly do the conversion.

Reply:
One example is the picking module - we take a slice of the picking buffer (`Uint8ClampedArray`) and assign it to `pickedColor` which is a `vec3`.
Before we can track down all use cases of `setUniforms`, it's hard to say how big the impact will be if we just reject malformed typed arrays. Handling it here seems to be the safest approach at the moment.